src/3d/qgsvirtualpointcloudentity_p.cpp

```diff
@@ -189,16 +223,20 @@ bool QgsVirtualPointCloudEntity::needsUpdate() const
 void QgsVirtualPointCloudEntity::updateBboxEntity()
 {
   QList<QgsAABB> bboxes;
-  const QVector<QgsPointCloudSubIndex> subIndexes = provider()->subIndexes();
-  for ( int i = 0; i < subIndexes.size(); ++i )
+  // we want to render bounding boxes only when zoomOutBehavior is RenderExtents or RenderOverviewAndExtents
+  if ( dynamic_cast<QgsPointCloudLayer3DRenderer *>( mLayer->renderer3D() )->zoomOutBehavior() != Qgis::PointCloudZoomOutRenderBehavior::RenderOverview )
```

Comment:
with dynamic_cast, we should check whether the resulting pointer is not null... if we're sure it's always a valid cast, then let's used static_cast...